lib/cylc/task_state.py

```diff
@@ -224,6 +233,14 @@ def satisfy_me(self, all_task_outputs):
                     self._is_satisfied = None
                     self._suicide_is_satisfied = None
 
+    def xtriggers_all_satisfied(self):
+        """Return True if xclock and all xtriggers are satisfied."""
+        if self.xclock is not None and not self.xclock[1]:
+            xclock_satisfied = False
+        else:
+            xclock_satisfied = True
+        return xclock_satisfied and all(self.xtriggers.values())
```

Comment:
```python
    return self.xclock and self.xclock[1] and all(self.xtriggers.values())
```

Reply:
ok...

Reply:
Addressed (slightly differently - "None and bool" returns None)